### ISSUE TITLE
Can't add Homebridge to Home app

### ISSUE THREAD

**Issue by nahueldl** [nfarina/homebridge#1966]
Like the tittle says I can't add homebridge to my Home app, I have already tried reinstalling every plugin and even homebridge itself but I can't make it work.
Also tried removing persist and accessories folders, and changing the username, pin and port number.

The plugins I have installed are:
- homebridge-config-ui-x
- homebridge-sonoff-tasmota-http

This is my current config.json file:
```json
{
	"bridge": {
		"name": "Homebridge",
		"username": "F4:8E:38:F4:62:0D",
		"port": 51826,
		"pin": "001-01-001"
	},
	"description": "test3",

	"platforms": [
		{
			"platform": "config",
			"name": "Config",
			"port": 8080,
			"sudo": false
		}
	],

	"accessories": [
		{
			"accessory": "SonoffTasmotaHTTP",
			"name": "Sonoff1",
			"hostname": "192.168.1.5"
		}

	]

}
```
I'm running homebridge on Windows 10 fully updated, and I've tried connecting through an iPhone 8 with iOS 12 beta 5 and through an iPhone 7 plus with iOS 11.4.1 .

This is an screenshot from the PowerShell
![PowerShell](https://image.ibb.co/jJB2RU/Screen_Homebridge.png)

And this one is from my phone, I'm sorry it's in spanish, it says something like it couldn't find the accessory and that I should make sure it is connected to my network. Both my pc and my phone are connected to the same network.
![PowerShell](https://image.ibb.co/fNq46U/4_D86179_B_F1_DB_44_C3_B0_A7_2_E8_E91_ED0073.png)

**Comment by brideone** [comment]
Hi! I don't know why this works (temporarily) for me, but try this. Use iphone's tethering and share your iPhone internet's connection with your homebridge server (windows 10 laptop/desktop). After that, try to add homebridge to your home's app. In my case it works and when I change my internet's connection again to my home's Wifi I lose my connection. I think maybe there's a problem related with the connection but I don't know what or why... Can you check it?

**Comment by nahueldl** [comment]
@brideone Unfortunately it didn't work for me

**Comment by brideone** [comment]
:( there are a lot of people like us, issues opened... hope the next release will fix the problem...

**Comment by Partizan7676** [comment]
Hi all! Exactly to post my info I register on Github. This and same issues when we can not add bridge, when it says accessories not ready, but in logs all is ok. I had that problem even on iOS 10(some same issues are opened after upgrade on iOS 11) and open ticket at Apple support but it was not solved. Problem is in... your Apple ID - it stores data about bridges, accessories you add to HomeKit and if you add and delete bridge several times you will get that error. Apple support can not reset your HomeKit data in iCloud or don't want?
How i find it. Bridge what I set up after adding camera show me only itself without cam in discover nearby devices. I rename camera and see it with bridge. And issue start - whatever I do, I receive msg that bridge is already added. After reading topics here I start changing username, pin and after that camera disappear and I see second, fake bridge, instead of camera. And name of fake bridge is known for me - few weeks ago I make experiments with some plugin and remember it's name. I recheck whole system to find where fake bridge can be, but nothing find. So I know exactly what I set - one bridge and one camera, but HomeKit app discover two bridges instead of one and camera. I take other iPhone iOS 11, but with other Apple ID and HomeKit app discover what I expect - my bridge and camera.
So problem is Apple ID. Just create new one for setting up your home. For operating just send invitation for yourself and other peoples.

**Comment by Partizan7676** [comment]
just give different name to bridge, change user, and pin. it helps me.

**Comment by pablos1rvent** [comment]
I have this problem too. Switched to new fresh hassbian with a new version of node and didn't work.

With 4 version of node worked, I don't know why...

**Comment by brideone** [comment]
@pablos1rvent I didn't solve it. I read forums everyday trying to find a solution. For the moment I'm without homebridge...

**Comment by nahueldl** [comment]
Done it @Partizan7676 and it didn't work. @brideone I'm doing the same thing every day. If I find a solution I'll post it here.

**Comment by brideone** [comment]
I show this thread https://github.com/nfarina/homebridge/issues/1952 and I think it's very interesting and It's maybe related with I said, something related with the network....

**Comment by pablos1rvent** [comment]
@brideone @nahueldl 

I have RPi and it didn't worked well. I reset pins and username, and it worked.

In windows 10 you need to check Firewall. Allow or disable firewall and try it. Check the router too.

**Comment by brideone** [comment]
I use a Mac mini as my homebridge server so... It's not the solution for me. It's so strange because it's the same problem in different OS. Today, if I have some time, I will try to investigate this https://github.com/nfarina/homebridge/issues/1952 Yesterday I tried to disable wifi connection and give ethernet connection to my Mac but it didn't work. I'll keep you informed guys!

**Comment by Partizan7676** [comment]
pablos1rvent, you do everything what don't need to do. i told to **change name of bridge**. Other like pin, username - not needed.

**Comment by pablos1rvent** [comment]
@Partizan7676 It worked for me... Maybe reinstalling the app solved it...

**Comment by pablos1rvent** [comment]
@Partizan7676 

> Two reasons why Homebridge may not be discoverable:
> 
> Homebridge server thinks it's been paired with, but iOS thinks otherwise. Fix: deleted persist/ directory which is next to your config.json.
> 
> iOS device has gotten your Homebridge username (looks like a MAC address) "stuck" somehow, where it's in the database but inactive. Fix: c**hange your username** in the "bridge" section of config.json to be some new value.

**Comment by brideone** [comment]
I tried https://github.com/nfarina/homebridge/issues/1952 but it didn't work for me. I still being able to use homebridge when I use other internet connection which is not my home's wifi connection. I use my default's orange router provided....

**Comment by nahueldl** [comment]
Finally I managed to make it work by following this issue #1957. I added the line `"mdns": {"interface": "192.168.1.14"},` "192.168.1.14"  is my laptop's local ip address. This forces the multicast lib to bind to an specify interface.
```json
{
	"mdns": {"interface": "192.168.1.14"},
	"bridge": {
		"name": "homebridge",
		"username": "22:34:56:78:9A:BC",
		"port": 51826,
		"pin": "031-45-154"
	},
	"description": "Prueba3",

	"platforms": [
		{
			"platform": "config",
			"name": "config",
			"port": 8080,
			"sudo": false
		}
	],

	"accessories": [
		{
			"accessory": "SonoffTasmotaHTTP",
			"name": "sonoff1",
			"hostname": "192.168.1.5"
		}

	]

}
```
Thank you all for your help, I wish this may help you.

**Comment by elfive1992** [comment]
@nahueldl sign in just to thank you.

**Comment by dyl4n20** [comment]
@nahueldl Worked 100% Thank you very much!

**Comment by bigbox486** [comment]
@nahueldl this also worked great for me. I signed in just to say thank you for spending the time troubleshooting and posting for the rest of us!!!

**Comment by ogiewon** [comment]
@nahueldl - Thank you!  This worked after many other attempts failed.

**Comment by MattR2424** [comment]
In case this doesn't work for some (it did for me the first time, not the second). You need to completely remove all of HomeBridge. Everything. Even node.js. I found that I was finally able to add HomeBridge after removing the entire folder itself, not just the files inside. This includes persist and accessories. After a fresh install, we're back up and running!!!

**Comment by henrik** [comment]
This is a bit silly, but after trying almost everything else (restarting the Mac running Homebridge, resetting Homebridge, regenerating username/PIN etc), what fixed it was... restarting my phone 🤦‍♂️

I *think* that after the restart, I opened Home.app, hit the "+", Add Accessory, "I Don't Have a Code or Cannot Scan", and then it showed me a list of accessories including Homebridge. I could just add it through there. But I'm pretty sure I tried that flow before and the restart is what did it.
